src/Persistence/MassTransit.Azure.Cosmos/Configuration/Configuration/CosmosSagaRepositoryConfigurator.cs

```diff
@@ -14,6 +14,13 @@ namespace MassTransit.Azure.Cosmos.Configuration
     using Saga.CollectionIdFormatters;
     using Saga.Context;
 
+    /// <summary>
+    /// A factory used to get instances of a <see cref="CosmosClient"/>
+    /// </summary>
+    /// <param name="cosmosClientName">The name of the client</param>
+    /// <param name="serializerSettings">Serializer settings for the client</param>
+    /// <returns>A <see cref="CosmosClient"/> used to interact with Cosmos DB</returns>
+    public delegate CosmosClient CosmosClientFactory(string cosmosClientName, JsonSerializerSettings serializerSettings);
```

Comment:
I'm not a big fan of putting delegates in the container, could this be an actual interface, `ICosmosClientFactory`?

Reply:
Added an interface with a basic implementation